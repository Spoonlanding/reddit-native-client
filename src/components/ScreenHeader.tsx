import * as React from 'react';
import {
  Header,
  Left,
  Button,
  Icon,
  Body,
  Title,
  Right,
} from 'native-base';

interface props {
  navigation: any;
}

const screenHeader: React.SFC<props> = ({ navigation }) => (
  <Header>
    <Left>
      <Button transparent >
        <Icon
          name="menu"
          onPress={() => navigation.navigate('DrawerOpen')}
        />
      </Button>
    </Left>
    <Body>
      <Title>{ navigation.state.routeName }</Title>
    </Body>
    <Right />
  </Header>
);

export default screenHeader;
