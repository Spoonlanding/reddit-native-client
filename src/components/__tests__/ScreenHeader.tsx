import 'react-native';
import * as React from 'react';
import { shallow } from 'enzyme';

import ScreenHeader from '../ScreenHeader';

describe('<ScreenHeader />', () => {
  let wrapper: any;
  const navigation: any = {
    navigate: jest.fn(),
    state: {
      routeName: 'foo',
    },
  };

  beforeEach(() => {
    wrapper = shallow(<ScreenHeader navigation={navigation} />);
  });

  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders routeName as title', () => {
    expect(wrapper.find('Styled(Title)').props().children).toEqual('foo');
  });

  it('opens drawer when menu icon is pressed', () => {
    wrapper.find({ name: 'menu' }).simulate('Press');
    expect(navigation.navigate).toHaveBeenCalledWith('DrawerOpen');
  });
});
