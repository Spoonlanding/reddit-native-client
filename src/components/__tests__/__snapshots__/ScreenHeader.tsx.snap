// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenHeader /> renders as expected 1`] = `
<Styled(Header)>
  <Styled(Left)>
    <Styled(Button)
      transparent={true}
    >
      <Styled(Icon)
        name="menu"
        onPress={[Function]}
      />
    </Styled(Button)>
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Title)>
      foo
    </Styled(Title)>
  </Styled(Body)>
  <Styled(Right) />
</Styled(Header)>
`;
